identity/src/main/java/com/android/identity/internal/Util.kt

```diff
@@ -0,0 +1,2209 @@
+        return cborEncode(cborBuildDateTime(timestamp))
+    }
+
+    /**
+     * Returns #6.0(tstr) where tstr is the ISO 8601 encoding of the given point in time.
+     * Only supports UTC times.
+     */
+    @JvmStatic
+    fun cborBuildDateTime(timestamp: Timestamp): DataItem {
+        val df = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX", Locale.US)
+        df.timeZone = TimeZone.getTimeZone("UTC")
+        val dateString = df.format(Date(timestamp.toEpochMilli()))
+        val dataItem: DataItem = UnicodeString(dateString)
+        dataItem.setTag(0)
+        return dataItem
+    }
+
+    @JvmStatic
+    fun cborDecode(encodedBytes: ByteArray): DataItem {
+        val bais = ByteArrayInputStream(encodedBytes)
+        val dataItems = try {
+            CborDecoder(bais).decode()
+        } catch (e: CborException) {
+            throw IllegalArgumentException("Error decoding CBOR", e)
+        }
+        require(dataItems.size == 1) {
+            "Unexpected number of items, expected 1 got ${dataItems.size}"
+        }
+        return dataItems.get(0)
+    }
+
+    @JvmStatic
+    fun cborDecodeBoolean(data: ByteArray): Boolean {
+        val simple: SimpleValue
+        simple = try {
+            cborDecode(data) as SimpleValue
+        } catch (e: ClassCastException) {
+            throw IllegalArgumentException("Data given cannot be cast into a boolean.", e)
+        }
+        return simple.simpleValueType == SimpleValueType.TRUE
+    }
+
+    /**
+     * Accepts a `DataItem`, attempts to cast it to a `Number`, then returns the value
+     * Throws `IllegalArgumentException` if the `DataItem` is not a `Number`. This
+     * method also checks bounds, and if the given data item is too large to fit in a long, it
+     * throws `ArithmeticException`.
+     */
+    @JvmStatic
+    fun checkedLongValue(item: DataItem): Long {
+        val bigNum = castTo(Number::class.java, item).value
+        val result = bigNum.toLong()
+        if (bigNum != BigInteger.valueOf(result)) {
+            throw ArithmeticException("Expected long value, got '$bigNum'")
+        }
+        return result
+    }
+
+    @JvmStatic
+    fun cborDecodeString(data: ByteArray): String {
+        return checkedStringValue(cborDecode(data))
+    }
+
+    /**
+     * Accepts a `DataItem`, attempts to cast it to a `UnicodeString`, then returns the
+     * value. Throws `IllegalArgumentException` if the `DataItem` is not a
+     * `UnicodeString`.
+     */
+    @JvmStatic
+    fun checkedStringValue(item: DataItem): String {
+        return castTo<UnicodeString, DataItem>(UnicodeString::class.java, item).string
+    }
+
+    @JvmStatic
+    fun cborDecodeLong(data: ByteArray): Long {
+        return checkedLongValue(cborDecode(data))
+    }
+
+    @JvmStatic
+    fun cborDecodeByteString(data: ByteArray): ByteArray {
+        val dataItem = cborDecode(data)
+        return castTo(
+            ByteString::class.java, dataItem
+        ).bytes
+    }
```

Comment:
can put `cborDecode(data)` inside `castTo` and use "equals" instead of `return`

```
    fun cborDecodeByteString(data: ByteArray) =
        castTo(ByteString::class.java, cborDecode(data)).bytes
```